src/NHibernate/Linq/Visitors/QueryModelVisitor.cs

```diff
@@ -527,10 +529,13 @@ private void AddJoin(JoinClause joinClause, QueryModel queryModel, bool innerJoi
 			var withClause = equalityVisitor.Visit(joinClause.InnerKeySelector, joinClause.OuterKeySelector);
 			var alias = _hqlTree.TreeBuilder.Alias(VisitorParameters.QuerySourceNamer.GetName(joinClause));
 			var joinExpression = HqlGeneratorExpressionVisitor.Visit(joinClause.InnerSequence, VisitorParameters);
+			var baseMemberCheker = new BaseMemberChecker(VisitorParameters.SessionFactory);
+
 			HqlTreeNode join;
-			// When associations are located inside the inner key selector we have to use a cross join instead of an inner
```

Comment:
A bit offtopic: Is this check still required in master? Or it also should be handled by #2361 and can be removed?

Reply:
No, it was obsolete by #2539. We should remove the `ParentJoinClause` property on the master branch as now is never set.